tools/resourcedocsgen/pkg/util/language/set.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package language
+
+import "iter"
```

Comment:
I'm not convinced we're getting a whole ton of value out of this over an array of languages/strings. In the use cases where we want all the languages, I think an array is fine and means we don't need `Iter()` etc. For the single use of `Has` we have that I think replaces `StringSet`, it looks like we could just write `if foo == "aws-native"` instead? While we might need flexibility later on, it doesn't seem like we do now/have for a while, and changing that lets us delete two whole files and avoid writing a ton of functionality. Even if we did want the set, I feel like we might still want to consider a more literal representation, unless we have good reasons for the bitset representation that I'm possibly missing?

Reply:
I think that `Set` has better copy semantics then `map[Language]struct{}`, which is why I chose it. I personally prefer that `All()` doesn't need to allocate, but its a needless performance optimization.

Given that `language.Set` always contains every language, I can rip it out. I'd rather have something like `language.Iter()` then `language.List()`, since it provides a cleaner `for lang := range language.Iter()` (no extra `_, `).

It was more important pre-refactor, when lots of functions were passing around lists of languages where they needed to apply.